src/renderers/dom/server/ReactMarkupChecksum.js

```diff
@@ -22,6 +22,13 @@ var ReactMarkupChecksum = {
    */
   addChecksumToMarkup: function(markup) {
     var checksum = adler32(markup);
+
+    // If root tag is a self-closing tag
+    if (markup.charAt(markup.length-2) === '/') {
+      return markup.substring(0, markup.length-2) + ' ' +
+        ReactMarkupChecksum.CHECKSUM_ATTR_NAME + '="' + checksum + '" />';
+    }
+    
```

Comment:
Instead of special casing the `/`, we can just take advantage of regular expression matches in `replace` (also I don't think we want to add the space before `/>` anyway)

```js
    return markup.replace(
      /\/?>/,
      ' ' + ReactMarkupChecksum.CHECKSUM_ATTR_NAME + '="' + checksum + '"$&'
    );
```